File: src/app/models/ThirdPartyProvider.js

// Define collection and schema for ThirdPartyProvider
export interface ThirdPartyProvider {
    name:
	type : string
    registrationId:
	type : string
    website:
	type : string
    Bank:
	type : Schema.Types.ObjectId
    Consents:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Consent' }]
#
    collection: 'thirdPartyProviders'
}

File: bankingOnGolang/internal/model/ThirdPartyProvider.go

package model

import (
)

//==============================================================
// ThirdPartyProvider Declaration
//==============================================================
type ThirdPartyProvider struct {
    BaseModel
     Name              string
    RegistrationId              string
    Website              string
    BankId           *uint
    Bank             *Bank `gorm:"foreignKey:BankId"`
     Consents             []Consent `gorm:"foreignKey:ConsentsFromThirdPartyProviderId"`

// parent associations as their child
    ThirdPartyProvidersFromBankId    *uint

}

